internal/kms/key_revocation.go

```diff
@@ -0,0 +1,33 @@
+package kms
+
+import (
+	"github.com/hashicorp/boundary/internal/kms/store"
+)
+
+type KeyRevocationStatus int
+
+const (
+	KeyRevocationStatusUnspecified KeyRevocationStatus = iota
+	KeyRevocationStatusPending
+	KeyRevocationStatusRunning
+	KeyRevocationStatusCompleted
+	KeyRevocationStatusFailed
+)
+
+func (k KeyRevocationStatus) String() string {
+	return [...]string{
+		"unspecified",
+		"pending",
+		"running",
+		"completed",
+		"failed",
+	}[k]
+}
+
+type KeyRevocation struct {
```

Comment:
it's a prototype... but I'm hopeful the implementation will have a factory for making these bits.